app/src/main/java/com/tomtom/sdk/examples/usecase/MapModeSelectionActivity.kt

```diff
@@ -0,0 +1,145 @@
+package com.tomtom.sdk.examples.usecase
+
+import android.content.Intent
+import android.graphics.PorterDuff
+import android.os.Bundle
+import android.view.Menu
+import android.view.MenuItem
+import android.view.View
+import android.widget.Button
+import android.widget.ImageButton
+import androidx.appcompat.app.AlertDialog
+import androidx.appcompat.app.AppCompatActivity
+import androidx.appcompat.widget.Toolbar
+import androidx.core.content.ContextCompat
+import com.google.android.material.card.MaterialCardView
+import com.tomtom.sdk.examples.R
+
+
+class MapModeSelectionActivity : AppCompatActivity() {
+    private var onlineButton: MaterialCardView? = null
+    private var offlineButton: MaterialCardView? = null
+    private var hybridButton: MaterialCardView? = null
+    private var mapMode = MapMode.ONLINE
+
+    companion object {
+        private const val STROKE_WIDTH = 4
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
```

Comment:
Quite long method, let's extract some logic to make it easier to understand

Reply:
done